### PR TITLE
[Autocomplete] implement `preload` option

### PR DESCRIPTION
**Pull request by seb-jean**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Tickets       | N/A
| License       | MIT

Added [preload](https://tom-select.js.org/docs/#preload) option to control the preload.
Default value is `false`

### CODE REVIEW

File: src/Autocomplete/assets/src/controller.ts [hunk 1]
```diff
@@ -216,4 +218,10 @@ export default class extends Controller {
     #dispatchEvent(name: string, payload: any): void {
         this.element.dispatchEvent(new CustomEvent(name, { detail: payload, bubbles: true }));
     }
+
+    get preload() {
+        if (this.preloadValue == 'false') return false;
+        if (this.preloadValue == 'true') return true;
+        return this.preloadValue;
```
Comment (weaverryan) [inline src/Autocomplete/assets/src/controller.ts:236]: Can you add some `{` to the `if` statements to standardize with the coding style? And also an empty break before the `return` statement. thx :)

File: src/Autocomplete/src/Form/AutocompleteChoiceTypeExtension.php [hunk 2]
```diff
@@ -97,12 +98,21 @@ public function configureOptions(OptionsResolver $resolver)
...
             return null === $options['autocomplete_url'] ? $value : false;
         });
+
+        $resolver->setNormalizer('preload', function (Options $options, $value) {
+            if (\is_bool($value)) {
+                $value = $value ? 'true' : 'false';
+            }
+
+            return $value;
+        });
```
Comment (weaverryan) [inline src/Autocomplete/src/Form/AutocompleteChoiceTypeExtension.php:115]: Would it work if we removed this and, instead, above, did:

```js
$values['preload'] = json_encode($options['preload']);
```

**Review by weaverryan — COMMENTED** [review]
This is very cool - thank you! Minor comments

File: src/Autocomplete/assets/src/controller.ts (hunk 1 above)
Comment (seb-jean) [reply src/Autocomplete/assets/src/controller.ts:236]: I changed the `if` statements.
I don't see where you want me to put the `break`.

File: src/Autocomplete/src/Form/AutocompleteChoiceTypeExtension.php (hunk 2 above)
Comment (seb-jean) [reply src/Autocomplete/src/Form/AutocompleteChoiceTypeExtension.php:115]: For boolean values it works, but not for strings

autocomplete

File: src/Autocomplete/assets/src/controller.ts (hunk 1 above)
Comment (weaverryan) [reply src/Autocomplete/assets/src/controller.ts:236]: Sorry, I meant an empty line break. You did it perfectly 👍

**Comment by weaverryan** [comment]
Thanks for this nice PR @seb-jean!

**Pull request merged**